### ISSUE TITLE
How to plot each element of mesh with colormap

### ISSUE THREAD

**Issue by nghiakvnvsd** [GLVis/glvis#116]
Hi everybody! 

I'm researching the topic, topology optimization. I have a question when I try to plot my results. That is how to plot each element of mesh with colormap (like the functions colormap() and imagesc() of Matlab). For examples, I have the mesh 180x60 and I perform the topology optimization problem I receive the array x_Phys(180x60). With matlab, I can do it by the following code:
`colormap(gray); imagesc(-x_Phys);` 
but with MFEM and GLVIS I do not know how to resolve it.  May you help me?

Thank you very much!!!

**Comment by v-dobrev** [comment]
You can create a `0`-order discontinuous space `L2_FECollection` and initialize a `GridFunction` on it with your data. In this case the `GridFunction` has one degree of freedom per mesh element. After that you can send it to GLVis as it is done in the MFEM examples, e.g. `ex1`/`ex1p`.

**Comment by nghiakvnvsd** [comment]
> You can create a `0`-order discontinuous space `L2_FECollection` and initialize a `GridFunction` on it with your data. In this case the `GridFunction` has one degree of freedom per mesh element. After that you can send it to GLVis as it is done in the MFEM examples, e.g. `ex1`/`ex1p`.

That is a good idea!! Thank you very much!

**Comment by nghiakvnvsd** [comment]
Sorry to bother you. I have done it but I do not know how to save my picture as file PDF. I have read the option of GLVIS and I saw the option "ctrl+P" but when I use it, file PDF does not have the figure.

**Comment by nghiakvnvsd** [comment]
One more thing, I try to use L2_FECollection but it does seem not work in 3D. my code is:
```
     FiniteElementCollection *fec_L2;
     fec_L2 = new L2_FECollection(1, dim);
     FiniteElementSpace *fespace_L2 = new FiniteElementSpace(mesh, fec_L2);
     GridFunction x_grid(fespace_L2);
     cout<<"x_grid.Size = "<<x_grid.Size()<<"\n";
```
It works very well when I use with dim = 2 but it returns the wrong output in 3D:

`x_grid.Size = 0`

**Comment by tzanio** [comment]
Take a screenshot with "Shift + S" instead.

**Comment by nghiakvnvsd** [comment]
It does not work. I try to with the example 1, it also returns the picture:

![image](https://user-images.githubusercontent.com/68639058/98635391-3d7b6500-2368-11eb-9f5b-3a2ed3fe4833.png)

**Comment by v-dobrev** [comment]
"Shift+S" should work. Can you provide some details: version of GLVis you are using, operating system, compiler, how you build GLVis, and any other information that you think may help us reproduce this issue.

**Comment by nghiakvnvsd** [comment]
> "Shift+S" should work. Can you provide some details: version of GLVis you are using, operating system, compiler, how you build GLVis, and any other information that you think may help us reproduce this issue.

I use GLVIS-3.4, ubuntu, gcc 7.5.0. I build Glvis by your instruction (https://mfem.org/building/). I try print the mesh of data mesh MFEM but it also does not work: 
In folder glvis I run: ./glvis -m ../mfem/data/beam-quad.mesh and then I "Shift+S". It returns the picture, GLVis_s01.png, in the glvis folder but do not show anything on the picture

**Comment by v-dobrev** [comment]
I was able to reproduce the issue but I'll need more time to figure out what is going wrong.

In the mean time, you can use the "Screenshot" app on Ubuntu to capture the GLVis window contents.

**Comment by tzanio** [comment]
@v-dobrev can you confirm that this is likely an issue with the graphics driver?

**Comment by v-dobrev** [comment]
The issue with taking screenshots (on Fedora) was due to the display driver: the open-source nvidia driver (https://nouveau.freedesktop.org/) had this issue; with the nvidia proprietary driver, the screenshots work just fine.

For ubuntu, I was able to reproduce the issue in a virtual machine where I can not test this solution (I can not change the driver) but I suspect the display driver is the issue as well.

As a workaround, it may be possible to capture the screenshots some other way (instead of using `glReadPixels`) but I have not looked into that.
